collector.go

```diff
@@ -137,18 +136,10 @@ var (
 )
 
 func NewExporter(connectionString string, namespace string, logger *slog.Logger) *Exporter {
-
-	db, err := getDB(connectionString)
```

Comment:
What do you think about validating the connection string so that we can flag errors before the scrape? Something like:

```go
import (
    "database/sql/driver"
    "github.com/lib/pq"
)

func validateConnectionStringFormat(connStr string) error {
    // Use pq driver's internal parsing
    connector, err := pq.NewConnector(connStr)
    if err != nil {
        return fmt.Errorf("invalid connection string format: %w", err)
    }
    _ = connector // We just wanted to validate parsing
    return nil
}
```